`win32_helpers.h` no longer compiles with `WIL_SUPPRESS_EXCEPTIONS` and C++20

Compile the following in C++20 mode:
```cpp
#define WIL_SUPPRESS_EXCEPTIONS
#include <Windows.h>
#include <wil/win32_helpers.h>

int main()
{

}
```

Observe error:
```
Build started at 03:35...
1>------ Build started: Project: ConsoleApplication10, Configuration: Debug x64 ------
1>ConsoleApplication10.cpp
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(213,21): error C2039: 'bit_cast': is not a member of 'std'
1>(compiling source file 'ConsoleApplication10.cpp')
1>    C:\Program Files (x86)\Windows Kits\10\Include\10.0.26100.0\ucrt\new.h(24,19):
1>    see declaration of 'std'
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(213,30): error C2062: type 'unsigned __int64' unexpected
1>(compiling source file 'ConsoleApplication10.cpp')
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(210,34): error C3615: constexpr function 'wil::filetime::to_int64' cannot result in a constant expression
1>(compiling source file 'ConsoleApplication10.cpp')
1>    Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(210,34):
1>    failure was caused by control reaching the end of a constexpr function
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(224,21): error C2039: 'bit_cast': is not a member of 'std'
1>(compiling source file 'ConsoleApplication10.cpp')
1>    C:\Program Files (x86)\Windows Kits\10\Include\10.0.26100.0\ucrt\new.h(24,19):
1>    see declaration of 'std'
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(224,30): error C2275: 'FILETIME': expected an expression instead of a type
1>(compiling source file 'ConsoleApplication10.cpp')
1>Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(221,38): error C3615: constexpr function 'wil::filetime::from_int64' cannot result in a constant expression
1>(compiling source file 'ConsoleApplication10.cpp')
1>    Z:\Throwaway\ConsoleApplication10\ConsoleApplication10\vcpkg_installed\x64-windows\x64-windows\include\wil\win32_helpers.h(221,38):
1>    failure was caused by control reaching the end of a constexpr function
1>Done building project "ConsoleApplication10.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build completed at 03:35 and took 00.784 seconds ==========
```

This seems to be caused by #490: inclusion of `<bit>` is now ultimately guarded by `WIL_USE_STL`, however further down the file, the feature check for `__cpp_lib_bit_cast` is done without checking `WIL_USE_STL` first. Should be easy to fix.

`WIL_USE_STL` seems to be automatically determined based on exception support, but the entire `<bit>` header is exception-free. Should we exclude exception-free usage of STL facilities from that `WIL_USE_STL` macro, or add something like `WIL_USE_NOEXCEPT_STL` to allow inclusion of exception-free STL bits? With the obvious fix (adding `WIL_USE_STL` to the feature checks), people who suppress exceptions are being pessimized (worse codegen, no constexpr filetime) for no valid reason.

--- a/include/wil/win32_helpers.h
+++ b/include/wil/win32_helpers.h
@@ -38,7 +38,7 @@
 #endif
 
 /// @cond
-#if __cpp_lib_bit_cast >= 201806L
+#if WIL_USE_STL && (__cpp_lib_bit_cast >= 201806L)
 #define __WI_CONSTEXPR_BIT_CAST constexpr
 #else
 #define __WI_CONSTEXPR_BIT_CAST inline
@@ -209,7 +209,7 @@ namespace filetime
 {
     constexpr unsigned long long to_int64(const FILETIME& ft) WI_NOEXCEPT
     {
-#if __cpp_lib_bit_cast >= 201806L
+#if WIL_USE_STL && (__cpp_lib_bit_cast >= 201806L)
         return std::bit_cast<unsigned long long>(ft);
 #else
         // Cannot reinterpret_cast FILETIME* to unsigned long long*
@@ -220,7 +220,7 @@ namespace filetime
 
     __WI_CONSTEXPR_BIT_CAST FILETIME from_int64(unsigned long long i64) WI_NOEXCEPT
     {
-#if __cpp_lib_bit_cast >= 201806L
+#if WIL_USE_STL && (__cpp_lib_bit_cast >= 201806L)
         return std::bit_cast<FILETIME>(i64);
 #else
         static_assert(sizeof(i64) == sizeof(FILETIME), "sizes don't match");

